components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/handlers/UsernameValidationConfigurationHandler.java

```diff
@@ -156,4 +160,50 @@ private String getUsernameRegEx(RealmConfiguration realmConfig) {
         return new String();
     }
 
+    /**
+     * Perform post actions after validation configuration are updated.
+     *
+     * @param tenantDomain      Tenant Domain.
+     * @param configuration     Updated validation configuration.
+     */
+    @Override
+    public void postValidationConfigurationUpdateHandler(String tenantDomain, ValidationConfiguration configuration)
+            throws InputValidationMgtServerException {
+
+        for (RulesConfiguration rule: configuration.getRules()) {
+            /*
+             Set required and support by default properties of the email attribute to true, when username field
+             validation configurations are set to emailFormatValidator.
+             */
+            if ("EmailFormatValidator".equals(rule.getValidatorName())) {
```

Comment:
Can the rule be null here? If so please check for null

Reply:
Addressed